### ISSUE TITLE
THREADS_WORKER_INIT_TIMEOUT environment variable

### ISSUE THREAD

**Issue by lsavage92** [andywer/threads.js#170]
Hello again! 👋

As far as I can tell this is an undocumented environment variable but I had to track it down due needing a longer timeout. Unfortunately in Vue applications we must specific our environment variables with a prefix of `VUE_APP_`. Because of this, I cannot increase the timeout of the worker init timeout without modifying the source of this library.

Is there anyway we can set which environment variable the library will look for?

**Comment by andywer** [comment]
Hey @lsavage92! Sorry, your question kind-of slipped my attention...

That’s a tricky one. In short: Right now you cannot, no.

I don’t know Vue too well. Is there maybe any way to opt-in to other env vars being processed when they don’t start with `VUE_APP_`?

I'm hesitant to check for another env var just for the Vue users. Does that concern every Vue app?

**Comment by lsavage92** [comment]
Hey thanks for responding! I think you are making the right move. I found recently that in the vue config file you can modify the environment like so:

```
module.exports = {
    chainWebpack: config => {
        
        const env = {
            // Notice the backticks AND the single quote. You need both.
            ENV_PROPERTY: `'ENV_VALUE'`,
            ENV_PROPERTY_2: `'${process.env.TEST}'` // Add value from existing env.
        }
        
        config.plugin('define')
            .tap(args => {
                Object.assign(args[0]['process.env'], env)
                return args
            })
    }
}```

This should allow you to assign an env var from `process.env` to a namespace other than one prefixed with `VUE_APP`.

**Comment by andywer** [comment]
Sounds good! Can I close the issue already or do you want to try it first?

On a different, but related note: The THREADS_WORKER_INIT_TIMEOUT variable might also be worth documenting.

**Comment by lsavage92** [comment]
Yes please close the issue! I have already confirmed that the above solution works. But I agree! That configuration can be useful!
